vgridjs/src/timeline_track.tsx

```diff
@@ -135,6 +135,80 @@ interface TimelineState {
   new_interval: Interval | null
 }
 
+interface TimelineNavigatorProps {
+  timeline_bounds: TimelineBounds,
+  timeline_width: number,
+  full_duration: number
+}
+
+@observer
+class TimelineNavigator extends React.Component<TimelineNavigatorProps, {}> {
+  private canvas_ref : React.RefObject<HTMLCanvasElement>;
+  private disposer : any;
+  
+  constructor (props: TimelineNavigatorProps) {
+    super(props);
+    this.canvas_ref = React.createRef();
+  }
+  
+  handleChange = (event: any) => {
+    if (event.target) {
+      let value  = event.target.valueAsNumber;
+      let max = parseInt(event.target.max);
+      let start = (this.props.full_duration * (value / this.props.timeline_width)) - (this.props.timeline_bounds.span()/2);
+      let end = (this.props.full_duration * (value / this.props.timeline_width)) + (this.props.timeline_bounds.span()/2);
+
+      if (start < 0) {
+        end -= start;
+        start = 0;
+      }
+      else if (end > this.props.full_duration) {
+        start -= end - this.props.full_duration;
+        end = this.props.full_duration
+      }
+      this.props.timeline_bounds.set_bounds(start, end);
+      this.forceUpdate();
+    }
+  }
+  
+  render_canvas() {
```

Comment:
Since we now have three different components that use this same canvas mobx hack, we should abstract this login into a component or mixin or something.